File: javascript/resistor-color-trio/resistor-color-trio.js
const COLORS = [
    "black", "brown", "red", "orange", "yellow",
    "green", "blue", "violet", "grey", "white"
];

export class ResistorColorTrio {
    constructor(bands) {
        let values = bands.map(b => {
            let value = COLORS.indexOf(b);

            if (value == -1)
                throw new "invalid color";
            else
                return value;
        });

        let total = (values[0] * 10 + values[1]) * 10 ** values[2];

        if (total % 1000 == 0)
            this.label = `Resistor value: ${total / 1000} kiloohms`;
        else
            this.label = `Resistor value: ${total} ohms`;
    }
}

File: lua/resistor-color-trio/resistor-color-trio.lua
COLORS = {
  black = 0,
  brown = 1,
  red = 2,
  orange = 3,
  yellow = 4,
  green = 5,
  blue = 6,
  violet = 7,
  grey = 8,
  white = 9
}

return {
  decode = function(c1, c2, c3)
    local value = (COLORS[c1] * 10 + COLORS[c2]) * 10 ^ COLORS[c3]

    if value < 1000 then
      unit = "ohms"
    else
      unit = "kiloohms"
      value = value / 1000
    end

    return value, unit
  end
}
